src/server/routes/entity/publisher.js

```diff
@@ -225,13 +224,13 @@ function transformNewForm(data) {
 	const relationships = entityRoutes.constructRelationships(
 		data.relationshipSection
 	);
-
 	return {
 		aliases,
 		areaId: data.publisherSection.area && data.publisherSection.area.id,
-		beginDate: dateObjectToISOString(data.publisherSection.beginDate),
+		beginDate: data.publisherSection.beginDate ?
+			dateObjectToISOString(data.publisherSection.beginDate) : '',
 		disambiguation: data.nameSection.disambiguation,
-		endDate: data.publisherSection.ended ?
+		endDate: data.publisherSection.endDate ?
```

Comment:
`endDate` is supposed to be an object of type `{day:string, month:string, year:string}`, although it sometimes is `false`. Not sure where this inconsistency comes from
This used to only try to parse and set the endDate if the ended checkbox is checked, which is probably still a useful check

Reply:
There will be a case when ended is true but endDate is null. i.e. when you know a publisher company has dissolved but don't know when.
Earlier the code was if ended is true, pick the endDate, and in some cases endDate was null and was creating some problems.

Reply:
Or we can have 2 checks ,'if ended is true and endDate is available', but I think having check for only endDate works as well

Reply:
I agree. The redux state takes care of erasing the endDate if you uncheck "ended", so you shouldn't see any case where ended is false but there is an endDate.
I tested that successfully.

Reply:
There is another underlying issue here. I've seen that endDate can either be a `{day, month, year}` object or a boolean. Not sure what are the conditions for there to be a boolean, but I doubt that it is correct.